generator/docs/NullOperator.md

```diff
@@ -0,0 +1,168 @@
+
+#### For illustrstion
+CSV files don't support representing the absence of a value differently to an empty string (unless all strings are always wrapped in quotes ([#441](https://github.com/ScottLogic/data-engineering-generator/pull/441)). 
+
+JSON files could be presented with `null` as the value for a property or excluding the property from the serialised result. This is the responsibility of the serialiser, and depends on the use cases.
+
+## The `null` operator and the `if` constraint
+With `if` constraints, the absence of a value needs to be considered in order to understand how the generator will behave. Remember, every set contains the empty set, unless excluded by way of the `not(is null)` constraint, for more details see [set restriction and generation](SetRestrictionAndGenerarion.md).
+
+Consider the following if constraint:
+
+```
+{
+    "if": {
+        {
+            "field": "field1",
+            "is": "equalTo",
+            "value": 5
+        }
+    },
+    "then": {
+        {
+            "field": "field2",
+            "is": "equalTo",
+            "value": "a"
+        }
+    }
+}
+```
+
+The generator will complete the `if` constraint as follows, to ensure the constraint is fully balanced:
+
+```
+{
+    "if": {
+        {
+            "field": "field1",
+            "is": "equalTo",
+            "value": 5
+        }
+    },
+    "then": {
+        {
+            "field": "field2",
+            "is": "equalTo",
+            "value": "a"
+        }
+    },
+    "else": {
+        {
+            "not": {
+                "field": "field1",
+                "is": "equalTo",
+                "value": 5
+            }
+        }
+    }
+}
+```
+
+This expression does not prevent the consequence (the `then` constraints) from being considered when `field1` has no value. Equally it does not say anything about the alternative consequence (the `else` constraints). As such both outcomes can be considered at any time.
+
+The solution to this is to express the `if` constraint as follows. This is not 'auto completed' for profiles as it would remove functionality that may be intended, it must be explicitly included in the profile.
+
+```
+{
+    "if": {
+        "allOf": [
+            {
+                "field": "field1",
+                "is": "equalTo",
+                "value": 5
+            },
+            {
+                "not": {
+                    "field": "field1",
+                    "is": "null"
+                }
+            }
+        ]
+    },
+    "then": {
+        {
+            "field": "field2",
+            "is": "equalTo",
+            "value": "a"
+        }
+    }
+}
+```
+
+The generator will complete the `if` constraint as follows, to ensure the constraint is fully balanced:
```

Comment:
What does it mean to complete a constraint?

Reply:
Wording updated to 

> The generator will expand the `if` constraint as follows, to ensure the constraint is fully balanced: